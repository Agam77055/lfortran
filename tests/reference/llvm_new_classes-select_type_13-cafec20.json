{
    "basename": "llvm_new_classes-select_type_13-cafec20",
    "cmd": "lfortran --no-color --show-llvm --new-classes {infile} -o {outfile}",
    "infile": "tests/../integration_tests/select_type_13.f90",
    "infile_hash": "8413deaa6aa9c492fa66fded9921b523764270cbc670d08a3a3c6ab0",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "llvm_new_classes-select_type_13-cafec20.stdout",
    "stdout_hash": "f80f5d352a8b5fa15af0f36e32c38bebe582b49f6c0c8e6720ee6017",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}