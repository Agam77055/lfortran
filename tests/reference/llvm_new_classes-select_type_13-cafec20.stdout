; ModuleID = 'LFortran'
source_filename = "LFortran"

%shape = type <{ i32 (...)** }>
%string_descriptor = type <{ i8*, i64 }>
%circle = type <{ %shape, float }>
%rectangle = type <{ %shape, float, float }>

@_ZTV20__si_class_type_info = external constant i8*
@_ZTV17__class_type_info = external constant i8*
@_Name_shape = private unnamed_addr constant [6 x i8] c"shape\00", align 1
@_Type_Info_shape = linkonce_odr unnamed_addr constant { i8*, i8* } { i8* bitcast (i8** getelementptr inbounds (i8*, i8** @_ZTV17__class_type_info, i64 2) to i8*), i8* getelementptr inbounds ([6 x i8], [6 x i8]* @_Name_shape, i32 0, i32 0) }, align 8
@_Name_circle = private unnamed_addr constant [7 x i8] c"circle\00", align 1
@_Type_Info_circle = linkonce_odr unnamed_addr constant { i8*, i8*, i8* } { i8* bitcast (i8** getelementptr inbounds (i8*, i8** @_ZTV20__si_class_type_info, i64 2) to i8*), i8* getelementptr inbounds ([7 x i8], [7 x i8]* @_Name_circle, i32 0, i32 0), i8* bitcast ({ i8*, i8* }* @_Type_Info_shape to i8*) }, align 8
@_VTable_circle = linkonce_odr unnamed_addr constant { [3 x i8*] } { [3 x i8*] [i8* null, i8* @_Type_Info_circle, i8* bitcast (float (%circle*)* @__module_select_type_13_module_circle_area to i8*)] }, align 8
@_Name_rectangle = private unnamed_addr constant [10 x i8] c"rectangle\00", align 1
@_Type_Info_rectangle = linkonce_odr unnamed_addr constant { i8*, i8*, i8* } { i8* bitcast (i8** getelementptr inbounds (i8*, i8** @_ZTV20__si_class_type_info, i64 2) to i8*), i8* getelementptr inbounds ([10 x i8], [10 x i8]* @_Name_rectangle, i32 0, i32 0), i8* bitcast ({ i8*, i8* }* @_Type_Info_shape to i8*) }, align 8
@_VTable_rectangle = linkonce_odr unnamed_addr constant { [3 x i8*] } { [3 x i8*] [i8* null, i8* @_Type_Info_rectangle, i8* bitcast (float (%rectangle*)* @__module_select_type_13_module_rectangle_area to i8*)] }, align 8
@_VTable_shape = linkonce_odr unnamed_addr constant { [3 x i8*] } { [3 x i8*] [i8* null, i8* @_Type_Info_shape, i8* null] }, align 8
@0 = private unnamed_addr constant [2 x i8] c"\0A\00", align 1
@string_const_data = private constant [21 x i8] c"Matched as rectangle\00"
@string_const = private global %string_descriptor <{ i8* getelementptr inbounds ([21 x i8], [21 x i8]* @string_const_data, i32 0, i32 0), i64 20 }>
@1 = private unnamed_addr constant [5 x i8] c"%s%s\00", align 1
@2 = private unnamed_addr constant [2 x i8] c"\0A\00", align 1
@string_const_data.1 = private constant [18 x i8] c"Matched as circle\00"
@string_const.2 = private global %string_descriptor <{ i8* getelementptr inbounds ([18 x i8], [18 x i8]* @string_const_data.1, i32 0, i32 0), i64 17 }>
@3 = private unnamed_addr constant [5 x i8] c"%s%s\00", align 1
@4 = private unnamed_addr constant [2 x i8] c"\0A\00", align 1
@serialization_info = private unnamed_addr constant [3 x i8] c"R4\00", align 1
@5 = private unnamed_addr constant [5 x i8] c"%s%s\00", align 1
@string_const_data.3 = private constant [11 x i8] c"ERROR STOP\00"
@string_const.4 = private global %string_descriptor <{ i8* getelementptr inbounds ([11 x i8], [11 x i8]* @string_const_data.3, i32 0, i32 0), i64 10 }>
@string_const_data.5 = private constant [2 x i8] c"\0A\00"
@string_const.6 = private global %string_descriptor <{ i8* getelementptr inbounds ([2 x i8], [2 x i8]* @string_const_data.5, i32 0, i32 0), i64 1 }>
@6 = private unnamed_addr constant [5 x i8] c"%s%s\00", align 1
@7 = private unnamed_addr constant [2 x i8] c"\0A\00", align 1
@string_const_data.7 = private constant [17 x i8] c"Matched as shape\00"
@string_const.8 = private global %string_descriptor <{ i8* getelementptr inbounds ([17 x i8], [17 x i8]* @string_const_data.7, i32 0, i32 0), i64 16 }>
@8 = private unnamed_addr constant [5 x i8] c"%s%s\00", align 1
@9 = private unnamed_addr constant [2 x i8] c"\0A\00", align 1
@string_const_data.9 = private constant [18 x i8] c"Matched as circle\00"
@string_const.10 = private global %string_descriptor <{ i8* getelementptr inbounds ([18 x i8], [18 x i8]* @string_const_data.9, i32 0, i32 0), i64 17 }>
@10 = private unnamed_addr constant [5 x i8] c"%s%s\00", align 1
@11 = private unnamed_addr constant [2 x i8] c"\0A\00", align 1
@string_const_data.11 = private constant [21 x i8] c"Matched as rectangle\00"
@string_const.12 = private global %string_descriptor <{ i8* getelementptr inbounds ([21 x i8], [21 x i8]* @string_const_data.11, i32 0, i32 0), i64 20 }>
@12 = private unnamed_addr constant [5 x i8] c"%s%s\00", align 1
@13 = private unnamed_addr constant [2 x i8] c"\0A\00", align 1
@serialization_info.13 = private unnamed_addr constant [6 x i8] c"R4,R4\00", align 1
@14 = private unnamed_addr constant [5 x i8] c"%s%s\00", align 1
@string_const_data.14 = private constant [11 x i8] c"ERROR STOP\00"
@string_const.15 = private global %string_descriptor <{ i8* getelementptr inbounds ([11 x i8], [11 x i8]* @string_const_data.14, i32 0, i32 0), i64 10 }>
@string_const_data.16 = private constant [2 x i8] c"\0A\00"
@string_const.17 = private global %string_descriptor <{ i8* getelementptr inbounds ([2 x i8], [2 x i8]* @string_const_data.16, i32 0, i32 0), i64 1 }>
@15 = private unnamed_addr constant [5 x i8] c"%s%s\00", align 1
@string_const_data.18 = private constant [11 x i8] c"ERROR STOP\00"
@string_const.19 = private global %string_descriptor <{ i8* getelementptr inbounds ([11 x i8], [11 x i8]* @string_const_data.18, i32 0, i32 0), i64 10 }>
@string_const_data.20 = private constant [2 x i8] c"\0A\00"
@string_const.21 = private global %string_descriptor <{ i8* getelementptr inbounds ([2 x i8], [2 x i8]* @string_const_data.20, i32 0, i32 0), i64 1 }>
@16 = private unnamed_addr constant [5 x i8] c"%s%s\00", align 1
@17 = private unnamed_addr constant [2 x i8] c"\0A\00", align 1
@string_const_data.22 = private constant [17 x i8] c"Matched as shape\00"
@string_const.23 = private global %string_descriptor <{ i8* getelementptr inbounds ([17 x i8], [17 x i8]* @string_const_data.22, i32 0, i32 0), i64 16 }>
@18 = private unnamed_addr constant [5 x i8] c"%s%s\00", align 1

define float @__module_select_type_13_module_circle_area(%circle* %this) {
.entry:
  %circle_area = alloca float, align 4
  %0 = getelementptr %circle, %circle* %this, i32 0, i32 1
  %1 = load float, float* %0, align 4
  %2 = fmul float 0x400921FA00000000, %1
  %3 = getelementptr %circle, %circle* %this, i32 0, i32 1
  %4 = load float, float* %3, align 4
  %5 = fmul float %2, %4
  store float %5, float* %circle_area, align 4
  br label %return

return:                                           ; preds = %.entry
  %6 = load float, float* %circle_area, align 4
  ret float %6
}

define float @__module_select_type_13_module_rectangle_area(%rectangle* %this) {
.entry:
  %rectangle_area = alloca float, align 4
  %0 = getelementptr %rectangle, %rectangle* %this, i32 0, i32 1
  %1 = load float, float* %0, align 4
  %2 = getelementptr %rectangle, %rectangle* %this, i32 0, i32 2
  %3 = load float, float* %2, align 4
  %4 = fmul float %1, %3
  store float %4, float* %rectangle_area, align 4
  br label %return

return:                                           ; preds = %.entry
  %5 = load float, float* %rectangle_area, align 4
  ret float %5
}

define i32 @main(i32 %0, i8** %1) {
.entry:
  call void @_lpython_call_initial_functions(i32 %0, i8** %1)
  %s1 = alloca %shape*, align 8
  store %shape* null, %shape** %s1, align 8
  %s2 = alloca %shape*, align 8
  store %shape* null, %shape** %s2, align 8
  %2 = call i8* @_lfortran_malloc(i64 12)
  call void @llvm.memset.p0i8.i32(i8* %2, i8 0, i32 12, i1 false)
  %3 = bitcast i8* %2 to %shape*
  store %shape* %3, %shape** %s1, align 8
  %4 = load %shape*, %shape** %s1, align 8
  %5 = bitcast %shape* %4 to %circle*
  %6 = bitcast %shape* %3 to i32 (...)***
  store i32 (...)** bitcast (i8** getelementptr inbounds ({ [3 x i8*] }, { [3 x i8*] }* @_VTable_circle, i32 0, i32 0, i32 2) to i32 (...)**), i32 (...)*** %6, align 8
  %7 = getelementptr %circle, %circle* %5, i32 0, i32 1
  %8 = getelementptr %circle, %circle* %5, i32 0, i32 0
  %9 = load %shape*, %shape** %s1, align 8
  %10 = bitcast %shape* %9 to i8*
  %11 = call i8* @__lfortran_dynamic_cast(i8* %10, i8* bitcast ({ i8*, i8*, i8* }* @_Type_Info_rectangle to i8*), i1 false)
  %12 = icmp ne i8* %11, null
  br i1 %12, label %then, label %else

then:                                             ; preds = %.entry
  %13 = load i8*, i8** getelementptr inbounds (%string_descriptor, %string_descriptor* @string_const, i32 0, i32 0), align 8
  call void @_lfortran_printf(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @1, i32 0, i32 0), i8* %13, i32 20, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @0, i32 0, i32 0), i32 1)
  br label %ifcont5

else:                                             ; preds = %.entry
  %14 = load %shape*, %shape** %s1, align 8
  %15 = bitcast %shape* %14 to i8*
  %16 = call i8* @__lfortran_dynamic_cast(i8* %15, i8* bitcast ({ i8*, i8*, i8* }* @_Type_Info_circle to i8*), i1 false)
  %17 = icmp ne i8* %16, null
  br i1 %17, label %then1, label %else4

then1:                                            ; preds = %else
  %18 = load i8*, i8** getelementptr inbounds (%string_descriptor, %string_descriptor* @string_const.2, i32 0, i32 0), align 8
  call void @_lfortran_printf(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @3, i32 0, i32 0), i8* %18, i32 17, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @2, i32 0, i32 0), i32 1)
  %19 = load %shape*, %shape** %s1, align 8
  %20 = bitcast %shape* %19 to %circle*
  %21 = getelementptr %circle, %circle* %20, i32 0, i32 1
  store float 1.000000e+01, float* %21, align 4
  %22 = load %shape*, %shape** %s1, align 8
  %23 = bitcast %shape* %22 to %circle*
  %24 = getelementptr %circle, %circle* %23, i32 0, i32 1
  %25 = load float, float* %24, align 4
  %26 = alloca float, align 4
  store float %25, float* %26, align 4
  %27 = call i8* (i8*, i64, i8*, i32, i32, ...) @_lcompilers_string_format_fortran(i8* null, i64 0, i8* getelementptr inbounds ([3 x i8], [3 x i8]* @serialization_info, i32 0, i32 0), i32 0, i32 0, float* %26)
  %28 = call i64 @_lfortran_str_len(i8* %27)
  %29 = call i8* @_lfortran_malloc(i64 16)
  %stringFormat_desc = bitcast i8* %29 to %string_descriptor*
  %30 = getelementptr %string_descriptor, %string_descriptor* %stringFormat_desc, i32 0, i32 0
  store i8* %27, i8** %30, align 8
  %31 = getelementptr %string_descriptor, %string_descriptor* %stringFormat_desc, i32 0, i32 1
  store i64 %28, i64* %31, align 4
  %32 = getelementptr %string_descriptor, %string_descriptor* %stringFormat_desc, i32 0, i32 0
  %33 = load i8*, i8** %32, align 8
  %34 = getelementptr %string_descriptor, %string_descriptor* %stringFormat_desc, i32 0, i32 1
  %35 = load i64, i64* %34, align 4
  %36 = trunc i64 %35 to i32
  call void @_lfortran_printf(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @5, i32 0, i32 0), i8* %33, i32 %36, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @4, i32 0, i32 0), i32 1)
  %37 = load %shape*, %shape** %s1, align 8
  %38 = bitcast %shape* %37 to %circle*
  %39 = getelementptr %circle, %circle* %38, i32 0, i32 1
  %40 = load float, float* %39, align 4
  %41 = fcmp une float %40, 1.000000e+01
  br i1 %41, label %then2, label %else3

then2:                                            ; preds = %then1
  %42 = load i8*, i8** getelementptr inbounds (%string_descriptor, %string_descriptor* @string_const.4, i32 0, i32 0), align 8
  %43 = load i8*, i8** getelementptr inbounds (%string_descriptor, %string_descriptor* @string_const.6, i32 0, i32 0), align 8
  call void (i8*, ...) @_lcompilers_print_error(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @6, i32 0, i32 0), i8* %42, i8* %43)
  call void @exit(i32 1)
  br label %ifcont

else3:                                            ; preds = %then1
  br label %ifcont

ifcont:                                           ; preds = %else3, %then2
  br label %ifcont5

else4:                                            ; preds = %else
  %44 = load i8*, i8** getelementptr inbounds (%string_descriptor, %string_descriptor* @string_const.8, i32 0, i32 0), align 8
  call void @_lfortran_printf(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @8, i32 0, i32 0), i8* %44, i32 16, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @7, i32 0, i32 0), i32 1)
  br label %ifcont5

ifcont5:                                          ; preds = %else4, %ifcont, %then
  %45 = call i8* @_lfortran_malloc(i64 16)
  call void @llvm.memset.p0i8.i32(i8* %45, i8 0, i32 16, i1 false)
  %46 = bitcast i8* %45 to %shape*
  store %shape* %46, %shape** %s2, align 8
  %47 = load %shape*, %shape** %s2, align 8
  %48 = bitcast %shape* %47 to %rectangle*
  %49 = bitcast %shape* %46 to i32 (...)***
  store i32 (...)** bitcast (i8** getelementptr inbounds ({ [3 x i8*] }, { [3 x i8*] }* @_VTable_rectangle, i32 0, i32 0, i32 2) to i32 (...)**), i32 (...)*** %49, align 8
  %50 = getelementptr %rectangle, %rectangle* %48, i32 0, i32 1
  %51 = getelementptr %rectangle, %rectangle* %48, i32 0, i32 2
  %52 = getelementptr %rectangle, %rectangle* %48, i32 0, i32 0
  %53 = load %shape*, %shape** %s2, align 8
  %54 = bitcast %shape* %53 to i8*
  %55 = call i8* @__lfortran_dynamic_cast(i8* %54, i8* bitcast ({ i8*, i8*, i8* }* @_Type_Info_circle to i8*), i1 false)
  %56 = icmp ne i8* %55, null
  br i1 %56, label %then6, label %else7

then6:                                            ; preds = %ifcont5
  %57 = load i8*, i8** getelementptr inbounds (%string_descriptor, %string_descriptor* @string_const.10, i32 0, i32 0), align 8
  call void @_lfortran_printf(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @10, i32 0, i32 0), i8* %57, i32 17, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @9, i32 0, i32 0), i32 1)
  br label %ifcont17

else7:                                            ; preds = %ifcont5
  %58 = load %shape*, %shape** %s2, align 8
  %59 = bitcast %shape* %58 to i8*
  %60 = call i8* @__lfortran_dynamic_cast(i8* %59, i8* bitcast ({ i8*, i8*, i8* }* @_Type_Info_rectangle to i8*), i1 false)
  %61 = icmp ne i8* %60, null
  br i1 %61, label %then8, label %else16

then8:                                            ; preds = %else7
  %62 = load i8*, i8** getelementptr inbounds (%string_descriptor, %string_descriptor* @string_const.12, i32 0, i32 0), align 8
  call void @_lfortran_printf(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @12, i32 0, i32 0), i8* %62, i32 20, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @11, i32 0, i32 0), i32 1)
  %63 = load %shape*, %shape** %s2, align 8
  %64 = bitcast %shape* %63 to %rectangle*
  %65 = getelementptr %rectangle, %rectangle* %64, i32 0, i32 1
  store float 5.000000e+00, float* %65, align 4
  %66 = load %shape*, %shape** %s2, align 8
  %67 = bitcast %shape* %66 to %rectangle*
  %68 = getelementptr %rectangle, %rectangle* %67, i32 0, i32 2
  store float 4.000000e+00, float* %68, align 4
  %69 = load %shape*, %shape** %s2, align 8
  %70 = bitcast %shape* %69 to %rectangle*
  %71 = getelementptr %rectangle, %rectangle* %70, i32 0, i32 1
  %72 = load float, float* %71, align 4
  %73 = alloca float, align 4
  store float %72, float* %73, align 4
  %74 = load %shape*, %shape** %s2, align 8
  %75 = bitcast %shape* %74 to %rectangle*
  %76 = getelementptr %rectangle, %rectangle* %75, i32 0, i32 2
  %77 = load float, float* %76, align 4
  %78 = alloca float, align 4
  store float %77, float* %78, align 4
  %79 = call i8* (i8*, i64, i8*, i32, i32, ...) @_lcompilers_string_format_fortran(i8* null, i64 0, i8* getelementptr inbounds ([6 x i8], [6 x i8]* @serialization_info.13, i32 0, i32 0), i32 0, i32 0, float* %73, float* %78)
  %80 = call i64 @_lfortran_str_len(i8* %79)
  %81 = call i8* @_lfortran_malloc(i64 16)
  %stringFormat_desc9 = bitcast i8* %81 to %string_descriptor*
  %82 = getelementptr %string_descriptor, %string_descriptor* %stringFormat_desc9, i32 0, i32 0
  store i8* %79, i8** %82, align 8
  %83 = getelementptr %string_descriptor, %string_descriptor* %stringFormat_desc9, i32 0, i32 1
  store i64 %80, i64* %83, align 4
  %84 = getelementptr %string_descriptor, %string_descriptor* %stringFormat_desc9, i32 0, i32 0
  %85 = load i8*, i8** %84, align 8
  %86 = getelementptr %string_descriptor, %string_descriptor* %stringFormat_desc9, i32 0, i32 1
  %87 = load i64, i64* %86, align 4
  %88 = trunc i64 %87 to i32
  call void @_lfortran_printf(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @14, i32 0, i32 0), i8* %85, i32 %88, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @13, i32 0, i32 0), i32 1)
  %89 = load %shape*, %shape** %s2, align 8
  %90 = bitcast %shape* %89 to %rectangle*
  %91 = getelementptr %rectangle, %rectangle* %90, i32 0, i32 1
  %92 = load float, float* %91, align 4
  %93 = fcmp une float %92, 5.000000e+00
  br i1 %93, label %then10, label %else11

then10:                                           ; preds = %then8
  %94 = load i8*, i8** getelementptr inbounds (%string_descriptor, %string_descriptor* @string_const.15, i32 0, i32 0), align 8
  %95 = load i8*, i8** getelementptr inbounds (%string_descriptor, %string_descriptor* @string_const.17, i32 0, i32 0), align 8
  call void (i8*, ...) @_lcompilers_print_error(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @15, i32 0, i32 0), i8* %94, i8* %95)
  call void @exit(i32 1)
  br label %ifcont12

else11:                                           ; preds = %then8
  br label %ifcont12

ifcont12:                                         ; preds = %else11, %then10
  %96 = load %shape*, %shape** %s2, align 8
  %97 = bitcast %shape* %96 to %rectangle*
  %98 = getelementptr %rectangle, %rectangle* %97, i32 0, i32 2
  %99 = load float, float* %98, align 4
  %100 = fcmp une float %99, 4.000000e+00
  br i1 %100, label %then13, label %else14

then13:                                           ; preds = %ifcont12
  %101 = load i8*, i8** getelementptr inbounds (%string_descriptor, %string_descriptor* @string_const.19, i32 0, i32 0), align 8
  %102 = load i8*, i8** getelementptr inbounds (%string_descriptor, %string_descriptor* @string_const.21, i32 0, i32 0), align 8
  call void (i8*, ...) @_lcompilers_print_error(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @16, i32 0, i32 0), i8* %101, i8* %102)
  call void @exit(i32 1)
  br label %ifcont15

else14:                                           ; preds = %ifcont12
  br label %ifcont15

ifcont15:                                         ; preds = %else14, %then13
  br label %ifcont17

else16:                                           ; preds = %else7
  %103 = load i8*, i8** getelementptr inbounds (%string_descriptor, %string_descriptor* @string_const.23, i32 0, i32 0), align 8
  call void @_lfortran_printf(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @18, i32 0, i32 0), i8* %103, i32 16, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @17, i32 0, i32 0), i32 1)
  br label %ifcont17

ifcont17:                                         ; preds = %else16, %ifcont15, %then6
  call void @_lpython_free_argv()
  br label %return

return:                                           ; preds = %ifcont17
  ret i32 0
}

declare void @_lpython_call_initial_functions(i32, i8**)

declare i8* @_lfortran_malloc(i64)

; Function Attrs: argmemonly nounwind willreturn writeonly
declare void @llvm.memset.p0i8.i32(i8* nocapture writeonly, i8, i32, i1 immarg) #0

declare i8* @__lfortran_dynamic_cast(i8*, i8*, i1)

declare void @_lfortran_printf(i8*, i8*, i32, i8*, i32)

declare i8* @_lcompilers_string_format_fortran(i8*, i64, i8*, i32, i32, ...)

declare i64 @_lfortran_str_len(i8*)

declare void @_lcompilers_print_error(i8*, ...)

declare void @exit(i32)

declare void @_lpython_free_argv()

attributes #0 = { argmemonly nounwind willreturn writeonly }
